### PR TITLE
Feature/papi 1306

### PR DESCRIPTION
**Pull request by Onebrownsound**
https://jwplayer.atlassian.net/browse/PAPI-1306. @gponickjw @amajedi @gtwohig @justiniso

### CODE REVIEW

**Comment by eduncan911** [comment]
@gtwohig should we rename this repository to: `jwplatform-demos` ?

**Comment by eduncan911** [comment]
LGTM

File: feed-formatter/feed_templates/aol.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <rss version="2.0"
```
Comment (gponickjw) [inline feed-formatter/feed_templates/aol.xml]: Calling this guy "AOL.xml" and having it be yahoo really makes my 90s self cry. I assume you imported these from somewhere?

File: feed-formatter/feeds.py [hunk 2]
```diff
@@ -4,72 +4,89 @@
 import datetime
 import os
 
-baseUrl      = 'http://content.jwplatform.com/feeds/%s.json'
+baseUrl = 'http://content.jwplatform.com/feeds/%s.json'
```
Comment (gponickjw) [inline feed-formatter/feeds.py]: I think our style guide (https://github.com/jwplayer/jwplatform/wiki/Media-Services-Style-Guide)

suggests we use 'http://content.jwplatform.com/feeds/{}.json'

File: feed-formatter/feeds.py [hunk 3]
```diff
@@ -4,72 +4,89 @@
...
-	if r.status_code == 200:
-		return r.json()
-	else:
-		raise Exception('Unable to fetch resource')
+    """
+
+    :param key: <string> Key of playlist/feed to retrieve
+    :return: <dict> JSON response
+    """
+    r = requests.get(baseUrl % key)
```
Comment (gponickjw) [inline feed-formatter/feeds.py]: See above, https://github.com/jwplayer/jwplatform/wiki/Media-Services-Style-Guide, suggests using {} and .format

File: feed-formatter/feeds.py [hunk 4]
```diff
@@ -4,72 +4,89 @@
...
+
+    :param json: <dict> Key:Value pairs to be injected
+    :param template_name: <string> Represents a particular template format found within feed_templates/
+    :return: <string> A string which represents the XML to returned in the response
+    """
+    try:
+        template = environment.get_template(template_name)
+        rendered = template.render(**json)
+    except:
+        raise Exception('Unable to render: %s' % template_name)
```
Comment (gponickjw) [inline feed-formatter/feeds.py]: Same feedback as above, .format

File: feed-formatter/feeds.py [hunk 5]
```diff
@@ -4,72 +4,89 @@
...
+    :param template_name: <string> Represents a particular template format found within feed_templates/
+    :return: <string> Rendered XML string with appropriate Key:Values injected into it.
+    """
+    try:
+        data = get(key)
+        parsed = parse(data)
+    except:
+        raise Exception('Unable to fetch and parse feed')
+    else:
+        rendered = toXML(parsed, "%s.xml" % template_name)
```
Comment (gponickjw) [inline feed-formatter/feeds.py]: ^

**Comment by gponickjw** [comment]
Some small cosmetic / formatting suggestions w/r/t our style guide.

That said - LGTM. One question I have that I didn't think worth pointing out on specific lines are defaults for params. In lots of places we use ='', should be using =None and nullchecking?

👍 LGTM otherwise

**Comment by Onebrownsound** [comment]
@gponickjw @gtwohig

**Comment by gponickjw** [comment]
LGTM 🍻 👽 🍊

**Comment by josiekeller** [comment]
@eduncan911 let's move forward with renaming the repo to jwplatform-demos, but let's sync up so I can modify the link on the dev site.

**Pull request merged**